CMakeLists.txt

```diff
@@ -99,6 +99,11 @@ set(IRODS_VERSION_PATCH "0")
 set(IRODS_VERSION "${IRODS_VERSION_MAJOR}.${IRODS_VERSION_MINOR}.${IRODS_VERSION_PATCH}")
 set(IRODS_PACKAGE_REVISION "0")
 
+# Bump this variable anytime the catalog schema is modified.
+set(IRODS_CATALOG_SCHEMA_VERSION "12")
+# Bump this variable anytime a JSON schema is modified.
+set(IRODS_CONFIGURATION_SCHEMA_VERSION "5")
```

Comment:
Consider documenting these in the developer docs?

Reply:
Open an issue for this.

Reply:
so this is replacing the version.json information?

did we capture the history from that file anywhere?

Reply:
No, they are not replacing/removing anything.

These variables collapse management of the versions to the CMakeLists.txt file just like the server version.

By changing these, all things which cared about these values get updated automatically. No one needs to touch the schemas or version.json.dist.in files anymore.

Reply:
okay, so we still will have version.json in 5.0

Reply:
Correct.

Reply:
Created #8021.

Reply:
The comments need to talk about major versions and breaking changes.